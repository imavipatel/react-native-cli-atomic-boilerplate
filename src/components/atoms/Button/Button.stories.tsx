import * as React from 'react';
import { storiesOf } from '@storybook/react-native';
import { action } from '@storybook/addon-actions';
import { Button } from './Button';
import CenterView from '../../../../storybook/utils/CenterView/CenterView';

export const actions = {
  onClick: action('onClick'),
};

storiesOf('Button', module)
  .addDecorator(story => <CenterView>{story()}</CenterView>)
  .add('default', () => <Button text="Button Text" />);
