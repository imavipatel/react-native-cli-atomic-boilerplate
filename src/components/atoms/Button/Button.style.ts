import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  buttonContainer: {
    minWidth: 200,
    minHeight: 60,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#08949B',
    borderStyle: 'solid',
    borderRadius: 20,
    borderWidth: 0,
  },
  buttonText: {
    color: 'white',
    fontWeight: '600',
    fontSize: 16,
  },
});
