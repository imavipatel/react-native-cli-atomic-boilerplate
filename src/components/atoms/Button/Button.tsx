import React from 'react';
import { Pressable, Text } from 'react-native';
import { styles } from './Button.style';

export interface ButtonProps {
  onPress?: () => void;
  text: string;
  testID: string;
}

export const Button: React.FC<ButtonProps> = props => {
  return (
    <Pressable
      onPress={props.onPress}
      style={styles.buttonContainer}
      testID={props.testID}>
      <Text style={styles.buttonText} testID="button-text">
        {props.text}
      </Text>
    </Pressable>
  );
};
