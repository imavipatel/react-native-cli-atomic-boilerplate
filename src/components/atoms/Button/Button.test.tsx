import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { Button } from './Button';

describe('Button', () => {
  const spyOnPress = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', () => {
    const tree = render(
      <Button testID="button" text="Button Text" onPress={spyOnPress} />,
    );

    expect(tree).toMatchSnapshot();
  });

  it('should call the provided onPress function when pressed', () => {
    const component = (
      <Button testID="button" text="Button Text" onPress={spyOnPress} />
    );
    const wrapper = render(component);

    fireEvent(wrapper.getByTestId('button'), 'onPress');
    expect(spyOnPress).toHaveBeenCalledTimes(1);
  });

  it('should render button', () => {
    const component = (
      <Button testID="button" text="Button Text" onPress={spyOnPress} />
    );
    const { getByTestId } = render(component);
    const button = getByTestId('button');
    expect(button).toBeDefined();
  });

  it('should render button text', () => {
    const component = (
      <Button testID="button" text="Button Text" onPress={spyOnPress} />
    );
    const { getByTestId } = render(component);
    const buttonText = getByTestId('button-text');
    expect(buttonText).toBeDefined();
  });

  it('should render title correctly', () => {
    const component = (
      <Button testID="button" text="Button Text" onPress={spyOnPress} />
    );
    const { getByText } = render(component);
    expect(getByText('Button Text')).toBeDefined();
  });
});
