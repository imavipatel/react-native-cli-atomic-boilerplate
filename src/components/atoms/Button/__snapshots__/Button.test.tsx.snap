// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#08949B",
      "borderRadius": 20,
      "borderStyle": "solid",
      "borderWidth": 0,
      "justifyContent": "center",
      "minHeight": 60,
      "minWidth": 200,
    }
  }
  testID="button"
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 16,
        "fontWeight": "600",
      }
    }
    testID="button-text"
  >
    Button Text
  </Text>
</View>
`;
